### PR TITLE
let's make a database/sql driver

### PR DESCRIPTION
**Pull request by alicebob**
Shouldn't be too hard to get a (readonly) database/sql driver working which supports basic selects. Right?

### CODE REVIEW

**Comment by vtolstov** [comment]
very cool!

File: driver/driver.go [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+func (*Tx) Commit() error {
+	return nil
+}
+
+// Statement is a single statement, belonging to a particular Connection.
+// It implements the driver.Stmt interface.
+type Statement struct {
+	dbh *db.Database
+	SQL string
+}
```
Comment (sbinet) [inline driver/driver.go]:
```suggestion
}

var (
    _ driver.Stmt = (*Statement)(nil)
)
```

just to make sure?

Comment (alicebob) [reply driver/driver.go]: Added, thanks!

**Comment by greenpau** [comment]
This is great! @alicebob , how far are you from the finish line on this one?

**Comment by alicebob** [comment]
On Wed, Apr 29, 2020 at 06:32:35PM -0700, Paul Greenberg wrote:
> This is great! @alicebob , how far are you from the finish line on this one?

I remember two problems:
- cancelling a query after fetching some, but not all, rows
- getting "unknown column" errors already in the sql.Query() call, not on the
  first Scan().

Both don't fit easily in the way sqlittle works. But they aren't impossible
to do either.

**Comment by alicebob** [comment]
This is Good Enough as a first version. Next up would be better support for data types (I guess mimic what the main sqlite driver does), and then maybe add some simple SQL features.

**Pull request merged**